src/fsharp/FSharp.Core/map.fs

```diff
@@ -558,7 +558,7 @@ namespace Microsoft.FSharp.Collections
             for (KeyValue(x,y)) in this do
```

Comment:
The iteration here could be improved, it currently allocates.  We could pass a byref through a recursive helper function, would be faster.  Not critical for this PR though.